bundles/org.eclipse.cdt.lsp/src/org/eclipse/cdt/lsp/LspPlugin.java

```diff
@@ -52,6 +53,7 @@ public void start(BundleContext context) throws Exception {
 		ServiceTracker<IWorkspace, IWorkspace> workspaceTracker = new ServiceTracker<>(context, IWorkspace.class, null);
 		workspaceTracker.open();
 		workspace = workspaceTracker.getService();
+		initialFileManager = new InitialFileManager(); // must be created before cLanguageServerProvider
```

Comment:
I would consider OSGi component instead and let Platform resolve order of construction for us.

Reply:
Good point. Via DI?

Reply:
I made the `InitialFileManager` singleton.

Reply:
I can suggest my changes after merging yours